code/controllers/subsystem/ghost_spawns.dm

```diff
@@ -260,6 +261,40 @@ SUBSYSTEM_DEF(ghost_spawns)
 
 	return TRUE
 
+/**
+ * Attempts to remove a signed-up mob from a poll.
+ *
+ * Arguments:
+ * * M - The mob to remove from the poll, if present.
+ * * Silent - If true, no messages will be sent to M about their removal.
+ */
+/datum/candidate_poll/proc/remove_candidate(mob/dead/observer/M, silent = FALSE)
+	. = FALSE
+	if(!istype(M) || !M.key || !M.client)
+		return
+	if(!(M in signed_up))
+		if(!silent)
+			to_chat(M, "<span class='warning'>You aren't signed up for this!</span>")
+		return
+
+	if(time_left() <= 0)
+		if(!silent)
+			to_chat(M, "<span class='danger'>It's too late to unregister yourself, selection has already begun!</span>")
+		return
+
+	signed_up -= M
+	if(!silent)
+		to_chat(M, "<span class='notice'>You have been unregistered as a candidate for this role. You can freely sign up again before the poll ends.</span>")
+
+		for(var/existing_poll in SSghost_spawns.currently_polling)
+			var/datum/candidate_poll/P = existing_poll
+			if(src != P && hash == P.hash && (M in P.signed_up))
+				P.remove_candidate(M, TRUE)
+	return TRUE
```

Comment:
Could you run me through the logic here, specifically why you're checking if `src` **isnt** `P`?

Reply:
This is the same logic in use in `/proc/sign_up()`, I just inverted the `(M in P.signed_up)` condition and set it to remove the candidate instead of signing them up.
From my understanding, we can have a few different polls in the same category (denoted by the hash) where if you sign up for one, you're signed up for all of them. This loop here similarly checks if you're signed up for one of those existing, same category polls and removes you from all of them. It passes silent = TRUE into the remove_candidate call to make sure the player doesn't get a ton of "you've been un-registered" messages.
The `src != P` check is there so we don't end up trying to remove M from this poll more than once, since we've already removed them a few lines up.